### PR TITLE
Remove the Unpooled.wrappedBuffer method that wraps a raw pointer

### PR DESCRIPTION
**Pull request by chrisvest**
Motiviation:
This method is inherently unsafe, and also appears to have no usage within Netty itself.

Modification:
Removed the Unpooled.wrappedBuffer method that takes a raw pointer, and also remove the WrappedUnpooledUnsafeDirectByteBuf that only existed to support this method.

Result:
Less unsafe code.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@chrisvest I think there was a reason I Introduced it ... Let me try to remember why...

**Comment by johnou** [comment]
> @chrisvest I think there was a reason I Introduced it ... Let me try to remember why...

for Agrona https://github.com/netty/netty/issues/5645

**Comment by chrisvest** [comment]
Not Agrona or Aeron themselves, but for an integration between those and Netty.

It can be worked around by tapping into `PlatformDependant`, or using reflection to call the relevant `DirectByteBuffer` constructor, or asking Agrona/Aeron to provide methods that do this.

The reason I want to get rid of unsafe APIs is that the JDK is itself increasingly placing restrictions on unsafe APIs. In that process, the JDK will be offering safer alternatives, and if we follow along with the development, we'll be better able to influence what those alternatives look like. We'll be less likely to get locked into an API that's hard to support down the line.
`ByteBuffer` will be a supported carrier type for native memory, probably forever. Meanwhile, being able to work with raw pointers in longs is something that'll probably require restricted APIs, and being able to access those will either require JVM command line flags, or require some sort of annotation in JMPS.

What do you guys think?

**Comment by normanmaurer** [comment]
I think as long as we will have some other way to do something similar with Netty in master it's fine.

**Comment by chrisvest** [comment]
@normanmaurer Okay. How about this?

File: buffer/src/main/java/io/netty/buffer/Unpooled.java [hunk 1]
```diff
@@ -213,9 +213,16 @@ public static ByteBuf wrappedBuffer(ByteBuffer buffer) {
     /**
      * Creates a new buffer which wraps the specified memory address. If {@code doFree} is true the
      * memoryAddress will automatically be freed once the reference count of the {@link ByteBuf} reaches {@code 0}.
+     * <p>
+     * <strong>Note:</strong> This method is inherently unsafe, and may not be supported on all platforms.
      */
     public static ByteBuf wrappedBuffer(long memoryAddress, int size, boolean doFree) {
-        return new WrappedUnpooledUnsafeDirectByteBuf(ALLOC, memoryAddress, size, doFree);
+        ByteBuffer buffer = PlatformDependent.directBuffer(memoryAddress, size);
```
Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/Unpooled.java:220]: This means it will fail on Java9+, correcT ?

Comment (chrisvest) [reply buffer/src/main/java/io/netty/buffer/Unpooled.java:220]: Yes, that seems to be correct.

**Pull request closed without merging**